daffodil-runtime1/src/main/scala/org/apache/daffodil/infoset/EXIInfosetInputter.scala

```diff
@@ -0,0 +1,127 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.daffodil.infoset
+
+import com.siemens.ct.exi.core.helpers.DefaultEXIFactory
+import javax.xml.transform.TransformerFactory
+import com.siemens.ct.exi.main.api.sax.EXISource
+import com.siemens.ct.exi.grammars.GrammarFactory
+import com.siemens.ct.exi.core.FidelityOptions
+import javax.xml.transform.sax.SAXSource
+import org.xml.sax.InputSource
+import java.io.ByteArrayInputStream
+import java.io.ByteArrayOutputStream
+import java.io.BufferedOutputStream
+import javax.xml.transform.stream.StreamResult
+import org.apache.daffodil.util.{ MaybeBoolean, Misc }
+import org.apache.daffodil.dpath.NodeInfo
+
+object EXIInfosetInputter {
+  def ConvertEXIToXML(input: java.io.InputStream): java.io.InputStream = {
+    val xsdLocation = Misc.getRequiredResource("org/apache/daffodil/xsd/XMLSchema.xsd")
+    lazy val exiFactory = DefaultEXIFactory.newInstance()
+    val grammarFactory = GrammarFactory.newInstance()
+    val g = grammarFactory.createGrammars(xsdLocation.toString)
+    exiFactory.setGrammars(g)
+    exiFactory.getFidelityOptions().setFidelity(FidelityOptions.FEATURE_PREFIX, true)
+    val saxSource = new EXISource(exiFactory)
+    val xmlReader = saxSource.getXMLReader()
+
+    val tf = TransformerFactory.newInstance()
+    val transformer = tf.newTransformer()
+
+    val exiSource = new SAXSource(new InputSource(input))
+    exiSource.setXMLReader(xmlReader);
```

Comment:
This is again first creating textual XML from the EXI, and then using the textual XML for input to daffodil unparse. 

We want to go directly from EXI data to daffodil unparse. 

You *could* do this via the SAXInfosetInputter, and given the symmetry with the SAXInfosetOutputter probably that should be done first.

For performance reasons, however, the natural control model for the unparser is not SAX (event calls), but more like StAX, which is pulling XML events. 

So we'll want to use a XML pull style API (i.e., StAX) if Exifficient provides one. In fact a generic StAXInfosetInputter that takes a StAX-API-Implementing object passed to its constructor would be the best way to do that leveraging the StAX standard.